src/pipeline-component-factory.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * File: pipeline-component-factory.js 
+ *
+ * Creates an RDF pipeline component factory. RDF pipeline components should instantiate with
+ * the following extensions beyond the standard noflo component:
+ *    - It must have an output and an error port.  Any other output ports are optional
+ *   - A vni facade as part of the node instance
+ *   - A wrapper object in the node instance that is to be used to hold a reference to the 
+ *     standard RDF framework wrapper functions: fRunUpdater, fExists, and fDestroyState. More 
+ *     details on wrappers can be found here: 
+ *        https://github.com/rdf-pipeline/noflo-rdf-pipeline/wiki/Wrapper-API
+ * In addition, rdf pipeline nodes should register an ondata callback function on each input port
+ * to be called whenever new data is received.
+ */
+
+var _ = require('underscore');
+var util = require('util');
+
+var nofloCompFactory = require('./noflo-component-factory');
+var promiseOutput = require('../src/promise-output');
+var vniManager = require('./vni-manager');
+
+/**
+ * Creates a noflo-component factory, as described by the nodeDef parameter, 
+ * customizing the factory with RDF pipeline specific data and funtionality. 
+ * 
+ * @param nodeDef a node definition describing the noflo component to be created
+ * @param wrapper an object representing the updater wrapper interface.  It will usually 
+ *        contain the 3 required updater wrapper functions: fRunUpdater, fExists, and 
+ *        fDestroyState.
+ * @param ondata (optional) event handler. If not provided, the framework-ondata will be
+ *        used.  Testing is a primary use case for this parameter.
+ * 
+ * @return a pipeline customized noflo component factory
+ */
+
+module.exports = function( nodeDef, wrapper, ondata ) { 
+
+    ondata = ondata || require('./framework-ondata');
```

Comment:
I suggest moving the require call to the top, like: var framework_ondata = require('./framework-ondata');